### DISCUSSION TITLE
How to get all the call transactions by func name of my contract?

### DISCUSSION THREAD

**Discussion by chlenc** [FuelLabs/fuel-indexer#707] [Q&A]
Hi guys
Have you seen our order book? swaynetwork.xyz
The name is not good, but we will rename it soon 
I ran into a problem with the indexing trades, and right now it works like that:
```
 #[storage(read)]
    fn trades(
        offset: u64,
    ) -> (Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>, Option<Trade>) {
        let mut vec = Vec::new();
        let mut i = 0;
        while i < 10 {
            let trade = if storage.trades.len() - 1 < i + offset {
                Option::None
            } else {
                storage.trades.get(storage.trades.len() - 1 - i - offset)
            };
            vec.push(trade);
            i += 1;
        }
        (
            vec.get(0).unwrap(),
            vec.get(1).unwrap(),
            vec.get(2).unwrap(),
            vec.get(3).unwrap(),
            vec.get(4).unwrap(),
            vec.get(5).unwrap(),
            vec.get(6).unwrap(),
            vec.get(7).unwrap(),
            vec.get(8).unwrap(),
            vec.get(9).unwrap(),
        )
    }
```

I believe this way is not perfect,
In the contract, we have func match_orders which checks orders and matches them about each other if it's possible
```
#[storage(read, write)]
fn match_orders(order_id_a: u64, order_id_b: u64); 
```

I guess will be the best way to make a chart like to fetch all of the successful match_orders calls and use outgoing payments and timestamps to make a chart
Also, this will provide less load on the network and the testnet will work faster
Can you explain or provide examples of how to fetch the match_orders calls, and get outgoing payments?

**Comment by ra0x3** [comment]
@chlenc Was this ever answered?

**Reply by chlenc** [reply]
We can use logs to avoid this ugly construction

**Comment by ra0x3** [comment]
@chlenc Awesome, I'll close this but feel free to open another discussion (and tag me personally) if you have any more questions :)
